File: API/Angular/src/app/viewmodels/GameViewModels/response-game-view-model.ts
import { ResponseRoundViewModel } from '../RoundViewModels/response-round-view-model';
import { ResponseUserViewModel } from '../UserViewModels/response-user-view-model';
import { ResponseUserGameViewModel } from '../UserGameViewModels/response-user-game-view-models';

export default class ResponseGameViewModel
{
  constructor(id, users, rounds, isOver, userGames)
  {
    this.id = id;
    this.users = users;
    this.rounds = rounds;
    this.isOver = isOver;
    this.userGames = userGames;
  }

  public id: number;
  public users: Array<ResponseUserViewModel>;
  public rounds: Array<ResponseRoundViewModel>;
  public userGames: Array<ResponseUserGameViewModel>;
  public isOver: boolean;
}

File: ViewModelsLayer/ViewModels/GameViewModels/ResponseGameViewModel.cs
﻿using System.Collections.Generic;
using ViewModelsLayer.ViewModels.UserViewModels;
using ViewModelsLayer.ViewModels.RoundViewModels;
using ViewModelsLayer.ViewModels.UserGameViewModels;

namespace ViewModelsLayer.ViewModels.GameViewModels
{
    public class ResponseGameViewModel
    {
        public int Id { get; set; }
        public List<ResponseUserViewModel> Users { get; set; }
        public List<ResponseRoundViewModel> Rounds { get; set; }
        public List<ResponseUserGameViewModel> UserGames { get; set; }
        public bool IsOver { get; set; }
    }
}
